lua/nvim-lsp-installer/servers/nimls/init.lua

```diff
@@ -0,0 +1,21 @@
+local server = require "nvim-lsp-installer.server"
+local shell = require "nvim-lsp-installer.installers.shell"
+
+return function(name, root_dir)
+  return server.Server:new {
+    name = name,
+    root_dir = root_dir,
+    homepage = "https://github.com/PMunch/nimlsp",
+    languages = { "nim" },
+    installer = {
+      shell.bash [[ 
+        git clone --depth 1 https://github.com/PMunch/nimlsp.git .
+        nimble build
+      ]]
+    },
+    default_options = {
+      cmd = { root_dir.."/nimlsp" },
+      filetypes = { "nim" },
```

Comment:
These options will essentially be merged with the lspconfig options, and [they already provide](https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/nimls.lua#L5) a `cmd` and `filetypes` for `nimls`. The only thing we need to do is extend the `PATH` environment, so that the client is able to locate the executable:

```suggestion
      cmd_env = {
          PATH = process.extend_path { root_dir },
      }
```